### ISSUE TITLE
corpus fail shuffle

### ISSUE THREAD

**Issue by humblemat810** [flairNLP/flair#2320]
**Describe the bug**
can be solved by using 
train_part_indices = list(range(len(self.corpus.train)))
or directly
self.corpus.train.indices
instead of 
train_part_indices = list(range(self.corpus.train))
![image](https://user-images.githubusercontent.com/67593116/123528901-005cbb00-d71e-11eb-89f5-2d68068b7b00.png)

**To Reproduce**
shuffle data using 
corpus = Corpus(train=train, test=test)

tars.add_and_switch_to_new_task('myTask', label_dictionary=corpus.make_label_dictionary())

trainer = ModelTrainer(tars, corpus)

trainer.train(base_path='resources/taggers/temp4', # path to store the model artifacts
              learning_rate=0.02, # use very small learning rate
              mini_batch_size=1, # small mini-batch size since corpus is tiny
              max_epochs=10, # terminate after 10 epochs
              train_with_dev=True,
              eval_on_train_fraction = 0.1,
              eval_on_train_shuffle=True
              )

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**
 - OS [e.g. iOS, Linux]:
 - Version [e.g. flair-0.3.2]:

**Additional context**
